### PR TITLE
[SPARK-23886][SS] Update query status for ContinuousExecution

### PR DESCRIPTION
**Pull request by gaborgsomogyi**
## What changes were proposed in this pull request?

Added query status updates to ContinuousExecution.

## How was this patch tested?

Existing unit tests + added ContinuousQueryStatusAndProgressSuite.

### CODE REVIEW

**Comment by gaborgsomogyi** [comment]
cc @jose-torres @HeartSaVioR @attilapiros

File: sql/core/src/main/scala/org/apache/spark/sql/streaming/StreamingQueryStatus.scala [hunk 1]
```diff
@@ -28,9 +28,10 @@ import org.apache.spark.annotation.InterfaceStability
  * Reports information about the instantaneous status of a streaming query.
  *
  * @param message A human readable description of what the stream is currently doing.
- * @param isDataAvailable True when there is new data to be processed.
+ * @param isDataAvailable True when there is new data to be processed. Doesn't apply
+ *                        to ContinuousExecution and always false.
```
Comment (attilapiros) [inline sql/core/src/main/scala/org/apache/spark/sql/streaming/StreamingQueryStatus.scala]: "and always false" => where it is always false. 
(Same for a few lines below)

**Review by attilapiros — APPROVED** [review]
Small suggestion regarding comment wording, besides that LGTM.

File: sql/core/src/main/scala/org/apache/spark/sql/streaming/StreamingQueryStatus.scala (hunk 1 above)
Comment (gaborgsomogyi) [reply sql/core/src/main/scala/org/apache/spark/sql/streaming/StreamingQueryStatus.scala]: Changed.

**Review by HeartSaVioR — APPROVED** [review]
+1 LGTM.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/continuous/ContinuousExecution.scala [hunk 2]
```diff
@@ -117,6 +117,7 @@ class ContinuousExecution(
     // For at least once, we can just ignore those reports and risk duplicates.
     commitLog.getLatest() match {
       case Some((latestEpochId, _)) =>
+        updateStatusMessage(s"Getting offsets from latest epoch $latestEpochId")
```
Comment (jose-torres) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/continuous/ContinuousExecution.scala]: nit: I'd mention that we're restarting the streaming query from an existing epoch in this message, to contrast with "starting new query" in the other branch

**Review by jose-torres — APPROVED** [review]
lgtm

File: sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/continuous/ContinuousExecution.scala (hunk 2 above)
Comment (gaborgsomogyi) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/continuous/ContinuousExecution.scala]: Fixed.

**Comment by gaborgsomogyi** [comment]
cc @tdas @zsxwing

**Comment by gaborgsomogyi** [comment]
cc @vanzin

**Comment by vanzin** [comment]
Looks good. I'd have chosen a shorter test name, but no biggie.

Merging to master.

**Pull request closed without merging**